packages/core/src/components/discord-input-text/DiscordInputText.ts

```diff
@@ -268,9 +287,21 @@ export class DiscordInputText extends LitElement {
 		}
 	}
 
+	private onModalClose() {
+		const rootDiscordMessagesElement = this.parentElement;
+		const dialogElement = rootDiscordMessagesElement?.shadowRoot?.querySelector('dialog');
+		if (dialogElement && !this.closeListener) {
+			dialogElement.addEventListener('close', () => {
```

Comment:
Instead of tracking this in a state use connectedCallback and disconnectedCallback to attach it. Better yet, use `@close` on the component itself.